src/main/java/org/dbflute/intro/app/web/playsql/data/PlaysqlDataAction.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2014-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+package org.dbflute.intro.app.web.playsql.data;
+
+import javax.annotation.Resource;
+
+import org.dbflute.intro.app.logic.exception.DirNotFoundException;
+import org.dbflute.intro.app.logic.playsql.data.PlaysqlDataLogic;
+import org.dbflute.intro.app.web.base.IntroBaseAction;
+import org.dbflute.intro.bizfw.tellfailure.OpenDirNotFoundException;
+import org.lastaflute.web.Execute;
+import org.lastaflute.web.response.JsonResponse;
+import org.lastaflute.web.servlet.request.ResponseManager;
+
+/**
+ * @author prprmurakami
+ */
+public class PlaysqlDataAction extends IntroBaseAction {
+
+    // ===================================================================================
+    //                                                                           Attribute
+    //                                                                           =========
+    @Resource
+    private PlaysqlDataLogic playsqlDataLogic;
+    @Resource
+    private ResponseManager responseManager;
+
+    // ===================================================================================
+    //                                                                             Execute
+    //                                                                             =======
+    @Execute(urlPattern = "{}/@word")
+    public JsonResponse<Void> open(String clientName) {
+        try {
+            playsqlDataLogic.openDataDir(clientName);
+        } catch (DirNotFoundException e) {
+            String playsqlDataDirPath = playsqlDataLogic.buildDataDirectoryPath(clientName);
```

Comment:
@jflute @EgumaYuto 
I changed the private method to a public method to get the directory path.
Is this okay?

Reply:
OK. (for business message)

Reply:
I fixed to get directory path from exceptions.

74a062b743fcaa698c62d205e1057b47d2584756